client/src/main/java/org/apache/rocketmq/client/impl/MQAdminImpl.java

```diff
@@ -410,19 +410,7 @@ public void operationComplete(ResponseFuture responseFuture) {
                     for (MessageExt msgExt : qr.getMessageList()) {
                         if (isUniqKey) {
                             if (msgExt.getMsgId().equals(key)) {
-
-                                if (messageList.size() > 0) {
-
-                                    if (messageList.get(0).getStoreTimestamp() > msgExt.getStoreTimestamp()) {
-
-                                        messageList.clear();
-                                        messageList.add(msgExt);
-                                    }
```

Comment:
If only one is to be kept, do we need to retain the original semantics and ensure that the storeTimestamp is the greatest for this one to be kept?

Reply:
Actually, A consumer that achieves idempotence only takes the first message, and ignores others. So I think it's not appropriate to keep the last message.